src/main/drivers/max7456.c

```diff
@@ -349,11 +367,21 @@ uint8_t max7456GetRowsCount(void)
     return (videoSignalReg & VIDEO_MODE_PAL) ? VIDEO_LINES_PAL : VIDEO_LINES_NTSC;
 }
 
+// When clearing the shadow buffer we fill with 0 so that the characters will
+// be flagged as changed when compared to the 0x20 used in the layer buffers.
 static void max7456ClearShadowBuffer(void)
 {
     memset(shadowBuffer, 0, maxScreenSize);
 }
 
+// Buffer is filled with the whitespace character (0x20)
+static void max7456ClearLayer(displayPortLayer_e layer)
+{
+    memset(getLayerBuffer(layer), 0x20, VIDEO_BUFFER_CHARS_PAL);
```

Comment:
Considering `0x00` is used by the hardware to clear the screen on initialisation, I think it makes more sense to use this character for the same purpose in the driver.
In a future iteration we could change all use of 0x20 in the drawing functions to 0x00 and free up one font character.

Reply:
Well a couple of things.
1. We don't make use of the hardware's ability to initialize the onboard buffer to 0 as there's no need.
2. We need a way to initialize the `shadowBuffer` to an unused value in cases where we want to force a full screen redraw. We currently do this by setting the `shadowBuffer` to 0x00 and since the foreground buffer is initialized to 0x20 we get a full refresh.
3. If we remapped any writes of the 0x20 character to 0x00 then yes I guess technically it frees up the 0x20 character, but we can never use it since any writes will translate to 0x00.

Reply:
1. to my understanding the MAX7456 does this on its own on power up - so using the 0x00 character for something that isn't whitespace will result in a screen filled with characters on power up;
2. fair enough - but I guess that this could be worked around by storing the fact that we want a full redraw in an external boolean;
3. we could map the code for the character that we are planning to use to store the font identification / classification to use 0x20 in the font, or store the identification / classification in 0x20 (which we can't do with 0x00 because of 1.

But mostly this is just my sense of reusing existing facilities speaking, which means it's not that important.

Reply:
1. Yes, you are correct, the MAX7456 does power up with the onboard buffer set to 0x00. So having this be a non-blank character is problematic. But I was also trying to point out that we don't use the built-in "clear" to reset it back to 0x00 in our logic but it's true that this doesn't change the initialization problem. For reference, the reason we don't use the "clear" is that the max7456 has no frame buffer. So the "clear" command results in the screen immediately going blank and then it takes time for us to stream out the actual elements. This results in a horrible flickering. I did some testing previously to see if the hardware clear could be useful but it was a dead-end.

2. Yes, an external flag to "force" a refresh is possible. But this is difficult in the current structure. The current implementation of the `drawScreen` has logic to only process a limited number of characters per call. So the drawing happens over multiple calls, picking up where it left off previously. So "force refresh" might get set with the screen partially drawn and then we'd have to persist it though the end of the current cycle and partway through the next full logical cycle and reset it at the appropriate time after we've logically processed through all the positions of the buffer.

3. This sounds like a good use for the 0x20 character. I'd guess we'd have to extend the max7456 driver to directly manipulate/retrieve this, but I guess doable. Using it for any other actual display character (like some symbol) would be problematic though.

So the main issue for me is 2. However I'm working on a follow-on to improve and optimize the `drawScreen` logic so I propose making the transition to using 0x00 for whitespace be part of that. I'm restructuring the `drawScreen` logic and the way it handles partial updates anyway so building in a "force refresh" won't be a problem. I can then simultaneously put in the logic to remap the whitespace writes to 0x00.

One other problem: using the 0x20 character for some metadata storage (like font ID) will be challenging from a firmware compatibility perspective. Once somebody uploads the font with that character populated with "data" then they'll never be able to use an earlier firmware version that doesn't have the 0x20 -> 0x00 remapping. It may force us to have "versioned" fonts and the ability to choose a "pre 4.2" type. Actually having fonts tied to a version could be helpful anyway as it always causes a problem if users neglect to update the font and we've had symbol changes. But just an additional layer we'd need to implement and support.

Reply:
Yeah, agreed that the 'versioned fonts' story has two sides. The unversioned fonts that we have at the moment mean that users can get into a situation where characters are changed in the firmware, and their OSD will look 'off' if they don't update to a font that matches, without any warning / indication to the user (i.e. 'it is just broken'). This will only get worse if we add support for things like custom pilot logos, or multiple font pages.
On the other hand, if we add support for versioned fonts, we'll need a considerable amount of logic to support this in the firmware.
Not sure which one of the two is better...